app/models/ability.rb

```diff
@@ -5,13 +5,21 @@ class Ability
 
   def initialize(user)
     user ||= User.new # guest user (not logged in)
-      if user.admin?
+      # if user.admin?
+      #   can :manage, :all
+      # else
+      #   can :manage, Article, User_id: user.id 
+      #   can :read, Article
+      #   can :read, Post
+      #   can :manage, Post, User_id: user.id 
+      # end
+
+      if user.has_role? :admin
         can :manage, :all
       else
-        can :manage, Article, User_id: user.id 
         can :read, Article
-        can :read, Post
-        can :manage, Post, User_id: user.id 
+        can :manage, Article if user.has_role?(:manager, Article)
+        can :write, Article, :id => Article.with_role(:moderator, user).pluck(:id)
```

Comment:
after plucking the id's.For what are using it?

Reply:
?

Reply:
I removed this for unwanted code.